itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -2237,4 +2242,27 @@ private void assertExternalFileInfo(List<String> expected, String dumplocation,
     }
     ReplicationTestUtils.assertExternalFileInfo(warehouseInstance, expected, externalTableInfoFile);
   }
+
+  private void assertFunctionJarsOnTarget(String functionName, List<String> expectedJars) throws IOException {
+    //correct location of jars on target is functionRoot/dbName/funcName/nanoTs/jarFile
+    FileStatus[] fileStatuses = replica.miniDFSCluster.getFileSystem()
+            .globStatus(
+               new Path(replica.functionsRoot + "/" + replicatedDbName.toLowerCase() + "/" +
+                       functionName.toLowerCase() + "/*/*"), path -> path.toString().endsWith("jar")
+            );
+    assertEquals(fileStatuses.length, expectedJars.size());
+    List<String> jars = Arrays.stream(fileStatuses).map(f -> {
+      String[] splits = f.getPath().toString().split("/");
+      return splits[splits.length - 1];
+    }).collect(Collectors.toList());
+    assertThat(jars, containsInAnyOrder(expectedJars.toArray()));
+
+    //confirm no jars in nested directories
+    FileStatus[] nestedFileStatuses = replica.miniDFSCluster.getFileSystem()
+            .globStatus(
+                    new Path(replica.functionsRoot + "/" + replicatedDbName.toLowerCase() + "/" + functionName + "/*/*/*")
+                    ,path -> path.toString().endsWith("jar")
+            );
+    assertEquals(nestedFileStatuses.length, 0);
```

Comment:
There should not be any directory itself right?

Reply:
Yes, i think a better check after matching the jar names will just be to confirm none of the files are directories. That will also be sufficient to test there are no nested directories. Will make this change.